core-lib/TestSuite/ExtensionTests.ns

```diff
@@ -0,0 +1,64 @@
+(* Testing SOMns' extension mechanism.
+
+   SOMns extensions are special jar files that contain primitives.
+   They are loaded like a normal module and behave the same.
+   Thus, from the perspective of the user, they are not distinguishable from
+   other modules.
+*)
+class ExtensionTests usingPlatform: platform testFramework: minitest = Value (
+| private TestContext  = minitest TestContext.
+  private ObjectMirror = platform mirrors ObjectMirror.
+  private FilePath     = platform files FilePath.
+
+  private system = platform system.
+
+  private jarName    = 'extension/build/test-extension.jar'.
```

Comment:
Will the extensions generally be in the same directory? If so, we could further hide the fact that these are written in Java and instead write the name as just `test-extension`. With this approach, it would be simple to swap out the Java module for a Newspeak module later on (without changing any code that uses the replaced module).

Reply:
Same directory is not necessary.
Why do we want to hide the fact that they are written in Java?
That would be a departure from what we currently do: https://github.com/smarr/SOMns/blob/release/core-lib/Platform.ns#L6
`vmMirror load: 'System.ns' nextTo: self`.

Not sure this is useful, and not sure it is desirable.
I would avoid unnecessary magic, I think.

Reply:
> Why do we want to hide the fact that they are written in Java?

It seemed like a good idea when I suggested it, but now that I'm trying to write something down I can't come up with an answer. Probably best we ignore this in that case :)